crates/spirv-attrib/src/lib.rs

```diff
@@ -1,16 +1,40 @@
-#![cfg_attr(target_arch = "spirv", feature(register_attr), register_attr(spirv))]
+use core::iter::FromIterator;
+use proc_macro::{Delimiter, Group, TokenStream, TokenTree};
 
-use proc_macro::TokenStream;
-use std::str::FromStr;
-
-#[cfg(not(target_arch = "spirv"))]
 #[proc_macro_attribute]
 pub fn spirv(_attr: TokenStream, item: TokenStream) -> TokenStream {
-    let mut item_str = item.to_string();
+    let mut tokens = Vec::new();
+    for tt in item {
+        match tt {
+            TokenTree::Group(group) => match group.delimiter() {
+                Delimiter::Parenthesis => {
+                    let mut sub_tokens = Vec::new();
+                    for tt in group.stream() {
+                        match tt {
+                            TokenTree::Group(group) => match group.delimiter() {
+                                Delimiter::Bracket => {
+                                    if group.stream().to_string().starts_with("spirv") {
+                                        sub_tokens.pop();
```

Comment:
You can check whether `sub_tokens.last()` is `TokenTree::Punct` of `'#'` at the same time you're checking whether the brackets have `spirv` inside, so that you don't handle anything that's not an attribute accidentally.

Reply:
done